app/models/manageiq/providers/google/cloud_manager/metrics_capture.rb

```diff
@@ -127,15 +126,19 @@ def perf_collect_metrics(_interval_name, start_time = nil, end_time = nil)
   #   aggregate metrics onto (will be modified by method)
   # @return nil
   def collect_metrics(google, start_time, end_time, schema, counter_values_by_ts)
+    interval = {
+      :start_time => start_time.to_datetime.rfc3339,
+      :end_time   => end_time.to_datetime.rfc3339
+    }
+
     schema[:google_metric_names].each do |google_metric_name|
-      options = {
-        :labels => schema[:target_to_google_labels].call(target),
-        :oldest => start_time.to_datetime.rfc3339,
-      }
+      # For filter creation and entity selection see https://cloud.google.com/monitoring/api/v3/filters
+      filter = "metric.type = \"#{google_metric_name}\" AND resource.labels.instance_id = \"#{target.ems_ref}\""
```

Comment:
> AND resource.labels.instance_id = \"#{target.ems_ref}\""

This looks like a much nicer interface than the target_to_google_labels

Reply:
It is indeed! Also the filter query allows much more precise matching and has a lot of options. That's really a nice thing. :+1: